### PR TITLE
timetables: code cleanup

### PR DESCRIPTION
**Pull request by ZoeBrunet**

### CODE REVIEW

**Comment by kinnou02** [comment]
First code PR after 4 days, that's way better than @pbougue !

File: source/time_tables/departure_boards.cpp [hunk 1]
```diff
@@ -117,18 +119,12 @@ void departure_board(PbCreator& pb_creator, const std::string& request,
     std::map<stop_point_route, vector_dt_st> map_route_stop_point;
 
     //Mapping route/stop_point
-    std::vector<stop_point_route> sps_routes;
+    boost::container::flat_set<stop_point_route> sps_routes;
```
Comment (xlqian) [inline source/time_tables/departure_boards.cpp:122]: :+1:

File: source/time_tables/departure_boards.cpp [hunk 2]
```diff
@@ -117,18 +119,12 @@ void departure_board(PbCreator& pb_creator, const std::string& request,
...
         const auto& jp = pb_creator.data.dataRaptor->jp_container.get(jpp.jp_idx);
         stop_point_route key = {jpp.sp_idx, jp.route_idx};
-        auto find_predicate = [&](stop_point_route spl) {
-            return spl.first == key.first && spl.second == key.second;
-        };
-        auto it = std::find_if(sps_routes.begin(), sps_routes.end(), find_predicate);
-        if(it == sps_routes.end()){
-            sps_routes.push_back(key);
-        }
+        sps_routes.insert(key);
```
Comment (xlqian) [inline source/time_tables/departure_boards.cpp:127]: minor: `sps_routes.insert({jpp.sp_idx, jp.route_idx});`

File: source/time_tables/departure_boards.cpp [hunk 3]
```diff
@@ -140,7 +136,7 @@ void departure_board(PbCreator& pb_creator, const std::string& request,
     // au meme couple (stop_point, route)
     // On veut en effet afficher les départs regroupés par route
     // (une route étant une vague direction commerciale
-    for(auto sp_route : sps_routes) {
+    for (const auto& sp_route: sps_routes) {
```
Comment (xlqian) [inline source/time_tables/departure_boards.cpp:139]: :+1:

**Pull request merged**